src/radar_control.py

```diff
@@ -681,6 +718,76 @@ def run_cfs_scan(
     return freq_data
 
 
+def cfs_block(ave_params, cfs_params_dict, cfs_sockets):
+    aveperiod = ave_params.aveperiod
+    if (
+        cfs_params_dict[aveperiod].last_cfs_set_time[aveperiod.beam_iter]
+        < time.time() - ave_params.aveperiod.cfs_stable_time
+        or aveperiod.cfs_always_run
+    ):
+        # Only let CFS run after the user set stable time has
+        # passed to prevent CFS from switching freqs to quickly
+        ave_params.sequence = aveperiod.cfs_sequence
+        ave_params.decimation_scheme = aveperiod.cfs_sequence.decimation_scheme
+
+        freq_spectrum = run_cfs_scan(
+            cfs_sockets[0],
+            cfs_sockets[1],
+            ave_params,
+            cfs_sockets[2],
```

Comment:
I cleaned it up a little bit. Kept the renaming of the list of sockets into variables at the top of `run_cfs_scan` so that its clear what those sockets are used for in that function.